src/dex/sundaeswap-v3.ts

```diff
@@ -0,0 +1,238 @@
+    };
+
+    constructor(requestConfig: RequestConfig = {}) {
+        super();
+
+        this.api = new SundaeSwapV3Api(this, requestConfig);
+    }
+
+    public async liquidityPoolAddresses(provider: BaseDataProvider): Promise<string[]> {
+        return Promise.resolve([this.poolAddress]);
+    }
+
+    async liquidityPools(provider: BaseDataProvider): Promise<LiquidityPool[]> {
+        const utxos: UTxO[] = await provider.utxos(this.poolAddress);
+
+        return await Promise.all(
+            utxos.map(async (utxo: UTxO) => {
+                return await this.liquidityPoolFromUtxo(provider, utxo);
+            })
+        )
+        .then((liquidityPools: (LiquidityPool | undefined)[]) => {
+            return liquidityPools.filter((liquidityPool?: LiquidityPool) => {
+                return liquidityPool !== undefined;
+            }) as LiquidityPool[];
+        });
+    }
+
+    public async liquidityPoolFromUtxo(provider: BaseDataProvider, utxo: UTxO): Promise<LiquidityPool | undefined> {
+        if (! utxo.datumHash) {
+            return Promise.resolve(undefined);
+        }
+
+        const relevantAssets: AssetBalance[] = utxo.assetBalances.filter((assetBalance: AssetBalance) => {
+            const assetBalanceId: string = assetBalance.asset === 'lovelace' ? 'lovelace' : assetBalance.asset.identifier();
+
+            return ! assetBalanceId.startsWith(this.validityTokenPolicyId);
+        });
+
+        // Irrelevant UTxO
+        if (relevantAssets.length !== 2) {
+            return Promise.resolve(undefined);
+        }
+
+        // Could be ADA/X or X/X pool
+        const assetAIndex: number = relevantAssets.length === 2 ? 0 : 1;
+        const assetBIndex: number = relevantAssets.length === 2 ? 1 : 2;
```

Comment:
I can't tell if `relevantAssets` is getting sorted somewhere, but they should be sorted lexicographically before setting the index for A and B.